src/libcore/char/methods.rs

```diff
@@ -121,15 +121,28 @@ impl char {
     #[stable(feature = "rust1", since = "1.0.0")]
     #[inline]
     pub fn to_digit(self, radix: u32) -> Option<u32> {
-        if radix > 36 {
-            panic!("to_digit: radix is too high (maximum 36)");
+        assert!(radix <= 36, "to_digit: radix is too high (maximum 36)");
+        if radix == 10 {
```

Comment:
When `radix` is constant, this is the same as the `radix < 10` case below (changed to `radix <= 10`), so I don't think a special case for 10 alone is necessary. See: https://godbolt.org/z/__H8mR

Reply:
@erikrose good catch, I would have thought that the `val < radix` check at the end would make a difference, but I just ran the numbers and they seem to be equivalent indeed. I've pushed another commit that removes the `radix == 10` special case.